src/Cli/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -32,10 +35,28 @@ public partial class RunCommand
         public bool NoLaunchProfile { get; private set; }
         private bool UseLaunchProfile => !NoLaunchProfile;
 
-        public int Execute()
+        public RunCommand(
+            bool noBuild,
+            string? projectFileOrDirectory,
+            string launchProfile,
+            bool noLaunchProfile,
+            bool noRestore,
+            bool interactive,
+            string[] restoreArgs,
+            string[] args)
         {
-            Initialize();
+            NoBuild = noBuild;
+            ProjectFileFullPath = DiscoverProjectFilePath(projectFileOrDirectory);
+            LaunchProfile = launchProfile;
+            NoLaunchProfile = noLaunchProfile;
+            Args = args;
+            RestoreArgs = GetRestoreArguments(restoreArgs);
```

Comment:
big change here: we precompute the RestoreArgs because we need them in more than one location now. Once for the implicit restore/build, if any, and once for the project evaluation to read the Run arguments. This is a behavior change/bugfix from previous SDKs, because there older SDKs wouldn't apply any properties so evaluation may differ.